crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -1775,6 +1781,23 @@ impl<'db> TypeInferenceBuilder<'db> {
                     lhs_value.clone() + rhs_value
                 }))
             }
+
+            (Type::StringLiteral(s), Type::IntLiteral(n), ast::Operator::Mult)
+            | (Type::IntLiteral(n), Type::StringLiteral(s), ast::Operator::Mult) => {
+                match usize::try_from(n) {
+                    Ok(n)
+                        if n * s.value(self.db).len()
```

Comment:
Is the multiplication here safe from overflow? Considering https://github.com/astral-sh/ruff/pull/13117/files#r1731915249 as well, I wonder if something like this might be better for this branch:

```rs
            (Type::StringLiteral(s), Type::IntLiteral(n), ast::Operator::Mult)
            | (Type::IntLiteral(n), Type::StringLiteral(s), ast::Operator::Mult) => {
                if n < 1 {
                    Type::StringLiteral(StringLiteralType::new(self.db, String::new()))
                } else if let Ok(n) = usize::try_from(n) {
                    if n.checked_mul(s.value(self.db).len())
                        .is_some_and(|new_length| new_length <= Self::MAX_STRING_LITERAL_SIZE)
                    {
                        let new_literal = s.value(self.db).repeat(n);
                        Type::StringLiteral(StringLiteralType::new(self.db, new_literal))
                    } else {
                        Type::LiteralString
                    }
                } else {
                    // This throws away the conversion error because we don't
                    // care: it just means the value was too large.
                    Type::LiteralString
                }
            }
```

Reply:
Ah, good call; I thought about overflow on the `try_from` but not the multiplication. Will update!